Bio/PDB/parse_pdb_header.py

```diff
@@ -129,18 +129,67 @@ def parse_pdb_header(infile):
     return _parse_pdb_header_list(header)
 
 
+def _parse_remark_465(line, out_dict):
+    """Parse missing residue remarks.
+
+    Fills two fields in out_dict: has_missing_residues and missing_residues.
+    It specification for REMARK 465 only gives templates, but does not say
+    they have to be followed. So we assume that not all pdb-files with a
+    REMARK 465 can be understood.
+
+    The dictionary entry "has_missing_residues" will be set to True, if
+    at least one REMARK 465 entry with non-empty text exists.
+
+    The dictionary entry 'missing_residues' will be filled with those
+    missing residues that can be parsed from the PDB header.
+    WARNING: The list out_dict['missing_residues'] will be
+    empty or incomplete, if the pdb-header is not successfully parsed.
+    """
+    if line:
+        out_dict["has_missing_residues"] = True
+        # Note that line has been stripped.
+        assert line[0] != " " and line[-1] not in "\n ", "line has to be stripped"
+    # Optional model number and residue name with 1 (e.g. for RNA) to 3 characters
+    modelnr_and_resname = "(\d+\s[\sA-Z][\sA-Z][A-Z]|[A-Z]?[A-Z]?[A-Z])"
+    chain = "\s([A-Za-z0-9])"
+    # Digit followed by optional insertion code.
+    # Note: Hetero-flags make no sense in contexty with missing residues.
+    ssseq = "\s+(\d+[A-Za-z]?)$"
+    pattern = modelnr_and_resname + chain + ssseq
+    match = re.match(pattern, line)
+    if match is not None:
+        residue = {}
+        if " " in match.group(1):
+            residue["model"], residue["res_name"] = match.group(1).split(" ")
+        else:
+            residue["model"] = None
+            residue["res_name"] = match.group(1)
+        residue["chain"] = match.group(2)
+        try:
+            residue["ssseq"] = int(match.group(3))
+        except:
+            residue["insertion"] = match.group(3)[-1]
+            residue["ssseq"] = int(match.group(3)[:-1])
+        else:
+            residue["insertion"] = None
+        out_dict["missing_residues"].append(residue)
+
+
 def _parse_pdb_header_list(header):
     # database fields
     dict = {'name': "",
```

Comment:
Oh yeah - I didn't see it, thanks. Definitely to be avoided for new code, and worth cleaning up where possible in old code (i.e. internal variable names can be changed, argument names cannot). Due to its usage being so common as shorthand for an identifier, trying to avoid built-in ``id`` seems like a lost battle.

In terms of pull requests, I'd prefer that kind of tidying up work be a separate pull request to ensure it gets done in a separate commit even if the pull request gets squashed.

@Bernhard10 if you want to fix that bit of the existing code, could you do it as a second pull request (after this one is merged to avoid any conflicts) please? Thanks!